.clang-format

```diff
@@ -0,0 +1,32 @@
+﻿---
+AlignConsecutiveMacros: 'true'
+AlignConsecutiveAssignments: 'true'
+AlignConsecutiveDeclarations: 'true'
+AlignTrailingComments: 'true'
+AllowAllConstructorInitializersOnNextLine: 'false'
+AllowAllParametersOfDeclarationOnNextLine: 'false'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortCaseLabelsOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: Inline
+AllowShortLoopsOnASingleLine: 'true'
+BinPackArguments: 'true'
+BinPackParameters: 'true'
+BreakBeforeBraces: Allman
+IndentWidth: '4'
+SpaceAfterCStyleCast: 'true'
+SpaceBeforeCpp11BracedList: 'true'
+SpaceBeforeParens: ControlStatements
+SpaceBeforeRangeBasedForLoopColon: 'true'
+SpaceInEmptyParentheses: 'false'
+SpacesInAngles: 'false'
+SpacesInCStyleCastParentheses: 'false'
+SpacesInContainerLiterals: 'false'
+SpacesInParentheses: 'false'
+Standard: Cpp11
```

Comment:
shouldn't it be `Cpp17` ? as we are using C++17

Reply:
Even tho i am using clang12 and the docs say cpp17 and even cpp20 is supported it will error as soon as im trying to format anything.

Reply:
I don't get it o_O

Reply:
<img src="https://i.imgur.com/5k471NQ.png">

Reply:
![image](https://user-images.githubusercontent.com/18329778/125205407-441dfb80-e2ac-11eb-98ec-060d91a4481b.png)

Reply:
According to https://zed0.co.uk/clang-format-configurator/ which Iraizo linked, this tool only has 

```
Format compatible with this standard, e.g. use A<A<int> > instead of A<A<int>> for LS_Cpp03.

Possible values:

LS_Cpp03 (in configuration: Cpp03) Use C++03-compatible syntax.

LS_Cpp11 (in configuration: Cpp11) Use features of C++11, C++14 and C++1z (e.g. A<A<int>> instead of A<A<int> >).

LS_Auto (in configuration: Auto) Automatic detection based on the input.
```

Reply:
So maybe this tool is not good